stdlib/digraph.mc

```diff
@@ -0,0 +1,169 @@
+include "option.mc"
+include "seq.mc"
+
+-- Represents a directed graph with labeled edges.
+
+-- Vertices and labels can be of any data types, as long as they have an
+-- equality function (given as arguments to the graph when created).
+
+-- An edge is represented as a triple (from, to, label), where the edge goes
+-- from "from" to "to", and "label" is the label of the edge. There can be
+-- several edges between two vertices.
+
+-- Vertices must be unique: there cannot be two vertices whose value of the
+-- equality function is true. This is maintained as invariant. Similarly, labels
+-- between any pair of vertices must be unique; this is also maintained as
+-- invariant.
+
+-- Presently, the graph is implemented using an adjacency map, which maps each
+-- vertex to a list of outgoing edges from that vertex.
+
```

Comment:
Suggestion: for readability, I would explicitly add the type of the graph. That is, something similar to
`type Digraph a = { adj : [(a,[(a,b)])], eqv : a -> a -> Bool, eql : b -> b -> Bool }` (not sure if the boot interpreter supports type variables yet though).

Reply:
Good idea. For now, added the type definition without the type parameters, i.e. just `type Digraph a = { adj : [(a,[(a,b)])], eqv : a -> a -> Bool, eql : b -> b -> Bool }`